wicketd/src/installinator_progress.rs

```diff
@@ -195,15 +204,44 @@ impl RunningUpdate {
     }
 
     fn send_and_next_state(
-        sender: mpsc::UnboundedSender<installinator_common::EventReport>,
+        log: &slog::Logger,
+        sender: watch::Sender<installinator_common::EventReport>,
         report: installinator_common::EventReport,
-    ) -> Self {
+    ) -> (Self, EventReportStatus) {
         let is_terminal = Self::is_terminal(&report);
-        _ = sender.send(report);
-        if is_terminal {
-            Self::Closed
-        } else {
-            Self::ReportsReceived(sender)
+        match sender.send(report) {
+            Ok(()) => {
+                if is_terminal {
+                    (Self::Closed, EventReportStatus::Processed)
+                } else {
+                    (
+                        Self::ReportsReceived(sender),
+                        EventReportStatus::Processed,
+                    )
+                }
+            }
+            Err(watch::error::SendError(report)) => {
+                // This typically means that the receiver is closed (typically
+                // due to the update process being aborted). If we enforced a
+                // 1:1 relationship between installinator and wicketd, this
+                // would indicate that the installinator should be aborted.
+                //
+                // Note that this "closed" is different from the Self::Closed
+                // state -- the latter indicates that the installinator has sent
+                // all of its messages.
+                slog::warn!(
+                    log,
+                    "progress receiver is closed -- marking aborted";
+                    "last_seen" => ?report.last_seen,
+                );
+                (
+                    // Don't set the state to Closed here even if this is a
+                    // terminal update, because we want to keep returning
+                    // ReceiverClosed.
```

Comment:
I don't entirely follow this comment, but I'm having a hard time finding where `Self::Closed` vs `Self::ReportsReceived(_)` is used or reported - can you point me in the right direction?

Reply:
I've expanded the comment -- hopefully this is clearer.

Reply:
It is, thank you!